src/ExpoShareIntentModule.types.ts

```diff
@@ -5,55 +5,96 @@ export type ChangeEventPayload = {
 export type StateEventPayload = {
   value: "pending" | "none";
 };
+/**
+ * The type of shared content that both Android and iOS have in common
+ */
+export type AndroidSharedMediaType = "file" | "text";
 
+/**
+ * The type of shared content that iOS supports.
+ */
+export type SharedMediaType = AndroidSharedMediaType | "media" | "weburl";
+
+/**
+ * Options for configuring the `useShareIntent` hook.
+ */
 export type ShareIntentOptions = {
+  /**
+   * If `true`, includes additional logs for debugging.
+   * @default false
+   */
   debug?: boolean;
+  /**
+   * If `true`, resets the shared content when the
+   * app goes into the background / foreground.
+   * @default true
+   */
   resetOnBackground?: boolean;
+  /**
+   * If `true`, disables shared intent.
+   * @default false
+   */
   disabled?: boolean;
+  /**
+   * Optional callback function that is triggered when the shared media resets.
+   */
   onResetShareIntent?: () => void;
 };
-
-export type ShareIntent = {
+/**
+ * Base type for what shared content is common between both platforms.
+ */
+interface BaseShareIntent {
+  text?: string | null;
+}
+/**
+ * Shared intent to represent both platforms.
+ */
+export interface ShareIntent extends BaseShareIntent {
   files: ShareIntentFile[] | null;
-  text: string | null;
+  type: SharedMediaType | null;
   webUrl: string | null;
-  type: "media" | "file" | "text" | "weburl" | null;
-};
+}
+/**
+ * Shared intent type for Android.
+ */
+export interface AndroidShareIntent extends BaseShareIntent {
+  files?: AndroidShareIntentFile[];
+  type: AndroidSharedMediaType;
+}
 
+/**
+ * Shared intent type for iOS.
+ */
+export interface IosShareIntent extends BaseShareIntent {
+  files?: IosShareIntentFile[];
+  type: SharedMediaType;
+}
+
+/**
+ * ShareIntentFile that is common among both platforms
+ */
 export interface ShareIntentFile {
-  path: string;
-  mimeType: string;
   fileName: string;
+  mimeType: string;
+  path: string;
   size: number | null;
 }
 
-export type IosShareIntent = {
-  text?: string;
-  files?: IosShareIntentFile[];
-  type: "media" | "file" | "text" | "weburl";
-};
-
 export interface IosShareIntentFile {
-  path: string;
-  type: string;
+  fileSize?: number; //TODO: Consolidate with size from ShareIntentFile
   fileName: string;
   mimeType: string;
-  fileSize?: number;
+  path: string; //TODO: Consolidate all 3 into filePath/path
+  type: string; //TODO: Eliminate? Duplicate with mimeType
```

Comment:
is not the same thing, ios module send a generic type as a number (see `IOS_SHARE_TYPE_MAPPING`)